netlify-templates/gatsby-ecommerce-theme
# Makes blog list title anchors and increase touch target

## Summary

In order to keep our site to be as accessible as possible, we want to make sure that all things that are links are still reachable for keyboard users. The solution for this right now is to make all the headings of the blog post previews anchor tags as opposed to using Read More. This will announce to the user the title at least and if they are using a screen reader they can still navigate to the rest of the text.

## Visual Differences

| Before | After |
| :---: | :---: |
| Mobile | |
| ![CleanShot 2022-05-18 at 15 14 58](https://user-images.githubusercontent.com/8431042/169138622-f805d04c-0bc3-4e1e-b72f-19c41982dae1.png) | ![CleanShot 2022-05-18 at 15 15 12](https://user-images.githubusercontent.com/8431042/169138665-4a390600-305a-461b-bac4-66ec105de51c.png) |
| Desktop* | |
| ![CleanShot 2022-05-18 at 15 15 59](https://user-images.githubusercontent.com/8431042/169138824-7438bbbd-7e1d-4848-8ff9-0cecefe7fa3c.png) | ![CleanShot 2022-05-18 at 15 16 18](https://user-images.githubusercontent.com/8431042/169138866-8ace2465-3967-4847-a28b-26180f55357a.png) |

* In the before of the desktop, there were no focus rings on the title elements. The touch targets were only the image and the title through `cursor: pointer` and `navigate()`. Now this is the entire card will be able for folk's to click on.

## Technical Detail

Previously, only the image or the title were ways for a user to navigate to the sample blog post. This could only be done through mouse users though. The elements were not anchors or focusable so if a keyboard user tried to focus on a blog element it wouldn't announce that it is navigable and couldn't actually be focused on.

To adjust this, we've made the title of each blog post an anchor but expanded the target to being the entire article card element. However, this will not apply the focus ring around the entire card, just the title. Definitely open to feedback on this and suggestions though. We've also removed the load more button to avoid any confusion.

## Manual Testing
- Visit the Deploy Preview
- Navigate to the Journal
- Verify that as you use <kbd>Tab</kbd> you are able to focus on all the titles within the blog preview list.
- Verify that if you hit <kbd>Enter</kbd> that this will navigate you to the `/blog/sample`
- Perform this same operation with the mouse again, clicking within any space of a card

## Relevant Issues and Documentation
- Closes #7 
- [Inclusive Components - Cards](https://inclusive-components.design/cards/)
- [Adrian Roselli - Block Links, Cards, Clickable Regions, Rows, etc.](https://adrianroselli.com/2020/02/block-links-cards-clickable-regions-etc.html)
